workers/executor/executors/legacy_executor.py

```
@@ -1169,16 +1185,28 @@ def _handle_index(self, context: ExecutionContext) -> ExecutionResult:
         except Exception as e:
             logger.error(
                 "Indexing failed: file=%s error=%s",
                 Path(file_path).name,
                 str(e),
             )
             status_code = getattr(e, "status_code", 500)
+            partial = []
+            if embedding is not None:
+                try:
+                    partial = list(embedding.flush_pending_usage())
+                except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)